### ISSUE TITLE
[java] BigIntegerInstantiation: False Positive with Java 17 and BigDecimal.TWO

### ISSUE THREAD

**Issue by Akromangel** [pmd/pmd#4886]
**Affects PMD Version:** 

7.0.0

**Rule:**

https://docs.pmd-code.org/pmd-doc-7.0.0/pmd_rules_java_performance.html#bigintegerinstantiation

**Description:**

The rule fails on Java 17.
BigDecimal.TWO was introduced in [Java 19](https://github.com/openjdk/jdk/commit/1d8e92ae0d2d0d6740e2171abef45545439e6414).

**Code Sample demonstrating the issue:**

```java
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
enum Ratio {
  TWO("One to Two", "1:2", new BigDecimal("2"));

    private final String title;
    private final String calculation;
    private final BigDecimal factor;
}
```

**Expected outcome:**

PMD reports a violation, but that's wrong. That's a false positive.

**Running PMD through:** *[Gradle]*
```sh
gradle :pmdMain
```

**Comment by mschaeferjelli** [comment]
This has re-emerged in 7.3.0.

**Comment by jsotuyod** [comment]
@mschaeferjelli we would need a lot more info on where you are seeing this… the tests we included with the original fix are still there and passing

https://github.com/pmd/pmd/blob/727279977dcb1598b3480fea1160e866094c79f2/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/performance/xml/BigIntegerInstantiation.xml#L178-L206

**Comment by mschaeferjelli** [comment]
Thanks for responding. I figured it out. The maven-pmd-plugin needed to have the JDK version explicitly configured.
```
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-pmd-plugin</artifactId>
                <version>3.24.0</version>
                <configuration>
                    <targetJdk>${java.version}</targetJdk>
                 ...
```
